### PR TITLE
HPCC-20150 Document ESP support for Dynamic ESDL Proxy mode

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>
@afishbeck Please review (Sent Pdf also)

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml [hunk 1]
```diff
@@ -1049,6 +1049,29 @@
           appropriate gateway's connection information, then use that
           information to create the actual gateway request connection.</para>
         </sect3>
+
+        <sect3>
+          <title>Configuring ESDL binding for Proxy Mode methods</title>
+
+          <para>You can specify a proxy for ESDL service methods on another
```
Comment (afishbeck) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml]: Perhaps something like "You can specify that ESDL service methods be proxied to another ESP instance"?

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml [hunk 2]
```diff
@@ -1049,6 +1049,29 @@
...
+          ESP instance. Set up the proxy in the dynamic configuration
+          associated with the dESDL service. </para>
+
+          <para>Under the Methods tag where you would add Method tags, you can
+          also add Proxy tags, as shown here:</para>
+
+          <programlisting>&lt;Methods&gt; 
+  &lt;Method name="myMethod" url="http://10.45.22.1:292/somepath"/&gt;
+  &lt;Method name="myMethod2" url="http://10.45.22.1:292/somepath"/&gt; 
+  &lt;Proxy method="myMethod3" address="http://10.45.22.1:292" /&gt;
```
Comment (afishbeck) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml]: "address" attribute was correct, but is changing to "forwardTo".

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml [hunk 3]
```diff
@@ -1049,6 +1049,29 @@
...
+          associated with the dESDL service. </para>
+
+          <para>Under the Methods tag where you would add Method tags, you can
+          also add Proxy tags, as shown here:</para>
+
+          <programlisting>&lt;Methods&gt; 
+  &lt;Method name="myMethod" url="http://10.45.22.1:292/somepath"/&gt;
+  &lt;Method name="myMethod2" url="http://10.45.22.1:292/somepath"/&gt; 
+  &lt;Proxy method="myMethod3" address="http://10.45.22.1:292" /&gt; 
+  &lt;Proxy method="myWild*" address="http://10.45.22.1:292" /&gt;
```
Comment (afishbeck) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml]: "address" attribute was correct, but is changing to "forwardTo".

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml [hunk 4]
```diff
@@ -1049,6 +1049,29 @@
...
+          <programlisting>&lt;Methods&gt; 
+  &lt;Method name="myMethod" url="http://10.45.22.1:292/somepath"/&gt;
+  &lt;Method name="myMethod2" url="http://10.45.22.1:292/somepath"/&gt; 
+  &lt;Proxy method="myMethod3" address="http://10.45.22.1:292" /&gt; 
+  &lt;Proxy method="myWild*" address="http://10.45.22.1:292" /&gt; 
+&lt;/Methods&gt;</programlisting>
+
+          <para>The Proxy tag also supports wildcards:</para>
+
+          <para><programlisting>&lt;Proxy method="myWild*" address="http://10.45.22.1:292" /&gt; </programlisting>This
```
Comment (afishbeck) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml]: "address" attribute was correct, but is changing to "forwardTo".

**Review by afishbeck — CHANGES_REQUESTED** [review]
@JamesDeFabia minor changes requested, otherwise looks good.

**Comment by JamesDeFabia** [comment]
@afishbeck Edits made, back to you

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml (hunk 2 above)
Comment (afishbeck) [reply docs/EN_US/HPCCClientTools/CT_Mods/CT_ESDL_CLI.xml]: @JamesDeFabia Looks like address has been changed to forwardTo below, but the one example above hasn't been changed.

**Review by afishbeck — CHANGES_REQUESTED** [review]
@JamesDeFabia just the one usage of "address" otherwise looks good.

**Review by afishbeck — APPROVED** [review]
@JamesDeFabia looks good.

**Comment by afishbeck** [comment]
@JamesDeFabia just tagging you.  not sure why this still shows up on my list.

**Comment by JamesDeFabia** [comment]
@richardkchapman this should be good to merge

**Pull request merged**